filters/openpolicyagent/opaauthorizerequest/opaauthorizerequest.go

```diff
@@ -143,8 +153,34 @@ func addRequestHeaders(fc filters.FilterContext, headers http.Header) {
 		}
 	}
 }
+func (f *opaAuthorizeRequestFilter) Response(fc filters.FilterContext) {
+	span, _ := f.opa.StartSpanFromFilterContext(fc)
+	defer span.Finish()
+
+	headersToAdd := fc.StateBag()[openpolicyagent.OpenPolicyAgentResponseHeadersKey]
+	if headersToAdd == nil {
+		return
+	}
 
-func (*opaAuthorizeRequestFilter) Response(filters.FilterContext) {}
+	var headers []*ext_authz_v3_core.HeaderValueOption
+	if headerValues, ok := headersToAdd.([]*ext_authz_v3_core.HeaderValueOption); ok {
+		headers = headerValues
+	} else {
+		errorMsg := fmt.Sprintf("Failed to read response headers: %v", headersToAdd)
+		f.opa.HandleInvalidDecisionError(fc, span, nil, errors.New(errorMsg), !f.opa.EnvoyPluginConfig().DryRun)
```

Comment:
use fmt.Errorf() instead of errors.New, then you can also drop the intermediate errorMsg var.

Reply:
Thanks for pointing out the possibility. Addressed.